scripts/docs/parse_assets.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Parse assets from directory and generate markdown files."""
+import generate_asset_documentation
+
+import argparse
+import re
+from pathlib import Path
+from typing import List
+
+import azureml.assets as assets
+import azureml.assets.util as util
+from azureml.assets.util import logger
+
+import snakemd
+from collections import defaultdict
+
+PARSED_COUNT = "parsed_count"
+
+
+
+def parse_assets(input_dirs: List[Path],
+                asset_config_filename: str,
+                pattern: re.Pattern = None):
+    """Parse all assets from input directory and generate documentation for each.
+
+    Args:
+        input_dirs (List[Path]): List of directories to search for assets.
+        asset_config_filename (str): Asset config filename to search for.
+        pattern (re.Pattern, optional): Regex pattern for assets to parse. Defaults to None.
+    """
+    asset_count = 0
+
+    references = defaultdict(list)
+
+    for asset_config in util.find_assets(input_dirs, asset_config_filename, pattern=pattern):
+        asset_count += 1
+
+        asset_type, asset_name, asset_file_name = generate_asset_documentation.create_asset_doc(asset_config.spec_with_path, asset_config.type)
+        references[asset_type].append((asset_name, asset_file_name))
+
+    logger.print(f"{asset_count} asset(s) copied")
+
+    for asset_type in references:
+
+        print(asset_type)
+        
+        # Create a new markdown file for each asset type
+        doc = snakemd.new_doc()
+        doc.add_heading(asset_type.capitalize() + "s", level=1)
+
+        # Create glossary that links to each asset of the asset type
+        doc.add_heading("Glossary", level=2)
+
+        for asset_name, asset_file_name in references[asset_type]:
+            doc.add_paragraph("[" + asset_file_name + "]: https://github.com/Azure/azureml-assets/wiki/" + asset_file_name)
+            
+        doc.add_horizontal_rule()
+
+        asset_links_list = []
+        for asset_name, asset_file_name in references[asset_type]:
+            asset_links_list.append("[" + asset_name + "][" + asset_file_name + "]")
```

Comment:
Assets can also be referenced directly by their md file names such as: [convert_model_to_mlflow](component-convert_model_to_mlflow)

Currently we're doing the following - just wanted to double-check if we want to keep this method?:
[component-convert_model_to_mlflow]: https://github.com/Azure/azureml-assets/wiki/component-convert_model_to_mlflow 
[convert_model_to_mlflow][component-convert_model_to_mlflow]